src/zcbor_encode.c

```diff
@@ -284,20 +284,24 @@ bool zcbor_bstr_start_encode(zcbor_state_t *state)
 }
 
 
-bool zcbor_bstr_end_encode(zcbor_state_t *state)
+bool zcbor_bstr_end_encode(zcbor_state_t *state, struct zcbor_string *result)
 {
 	const uint8_t *payload = state->payload;
 	struct zcbor_string value;
 
+	if (result == NULL) {
+		result = &value;
```

Comment:
This takes the address of a temporary object, it might not be used by the calling function but shouldn't be returned/updated to it

Reply:
The result is not being dereferenced, so the address to the temporary object is not passed back to the caller.

Reply:
This looks so much like a bug at a first glance that I would spent a comment here so that future me is not confused again.

Reply:
Added